src/core/public/core_system.ts

```diff
@@ -200,6 +223,17 @@ export class CoreSystem {
       });
       const uiSettings = await this.uiSettings.start();
 
+      application.registerMountContext(this.coreContext.coreId, 'core', () => ({
```

Comment:
I don't think anything during `setup` should be callable during any handlers.

For the fatalErrors example, I plan to not expose this to plugins at all and replace it with an ApplicationService-level API that only renders UI in the application container rather than over the entire page: https://github.com/elastic/kibana/issues/41980

Plugins can still create a fatal error by throwing an error during their `setup` or `start` functions, which will automatically be handled by Core.